File: Tutorial4-5SubComponentBuilders/src/main/java/com/example/tutorial4_5subcomponentbuilders/di/ApplicationComponent.java
package com.example.tutorial4_5subcomponentbuilders.di;


import javax.inject.Singleton;

import dagger.Component;

// This is parent component
@Component(modules = ApplicationModule.class)
@Singleton
public interface ApplicationComponent {

    // Calls SubComponent Builder from MainActivity using ApplicationComponent
    ToastMakerSubComponent.Builder toastMakerBuilder();
}

File: Tutorial4-5SubComponentBuilders/src/main/java/com/example/tutorial4_5subcomponentbuilders/di/ApplicationComponent.kt
package com.example.tutorial4_5subcomponentbuilders.di

import dagger.Component
import javax.inject.Singleton

// This is parent component
@Component(modules = [ApplicationModule::class])
@Singleton
interface ApplicationComponent {

    // Calls SubComponent Builder from MainActivity using ApplicationComponent
    fun toastMakerBuilder(): ToastMakerSubComponent.Builder

}